src/debug_menu.cpp

```diff
@@ -570,6 +570,100 @@ static void monster_ammo_edit( monster &mon )
     }
 }
 
+static std::string query_npctalkvar_new_value()
+{
+    std::string value;
+    string_input_popup popup_val;
+    popup_val
+    .title( _( "Value" ) )
+    .width( 85 )
+    .edit( value );
+    return value;
+}
+
+static void edit_global_npctalk_vars()
+{
+    uilist global_var_list;
+    global_var_list.desc_enabled = true;
+    global_var_list.title = _( "Edit npctalkvar variables (global)" );
+    global_variables &globvars = get_globals();
+    // some ordering shennanigans so that i == 0 is the option to add a new var
+    int i = 1;
+    std::vector<std::string> keymap_index = {_( "Add new npctalkvar (global)" )};
+    global_var_list.addentry_desc( 0, true, input_event(), keymap_index[0], "" );
+    for( std::pair<const std::string, std::string> &some_global : globvars.get_global_values() ) {
+        keymap_index.emplace_back( some_global.first );
+        std::string description = string_format( _( "raw var value: %s" ), some_global.second );
+        // only run std::stoi if these are actually doubles and won't explode us
+        if( strtol( some_global.second.c_str(), NULL, 0 ) ) {
+            description += "\n";
+            description += string_format( _( "as time_duration: %s" ),
+                                          to_string( time_duration::from_turns( std::stof( some_global.second ) ) ) );
+            description += "\n";
+            description += string_format( _( "as time_point: %s" ),
+                                          to_string( calendar::turn_zero + time_duration::from_turns( std::stof( some_global.second ) ) ) );
```

Comment:
There's `try_parse_integer` that'll be much safer than using the C standard library functions for parsing.
https://github.com/CleverRaven/Cataclysm-DDA/blob/9857702e32ae7c56fe49abfaf6fa0e8d050814f0/src/try_parse_integer.h#L7-L16

(You're using stof, but the functions you give the results don't care about floats).

Reply:
Well, I guess there WAS a better way. I'll push that in a bit

FYI I'm using std::stof because of #65522 - std::stoi parses very large values like `5.56e+06` as merely `5.56`. That was a very nasty thing to learn

Reply:
That's horrifying :)

Reply:
I don't think try_parse_integer likes our doubles stored as strings :(

![image](https://github.com/user-attachments/assets/2de0d63b-7bd7-479c-a2c9-64375505420e)

Reply:
I see. I guess the scientific notation is only for floating point numbers. Someone could create a `try_parse_float` to solve this, but it would require some research. https://github.com/CleverRaven/Cataclysm-DDA/pull/48431 for context.